cmd/cmd.go

```diff
@@ -721,6 +722,188 @@ func checkServerHeartbeat(_ *cobra.Command, _ []string) error {
+	}
+
+	// Derive the manifest path
+	manifestPath, err := mp.GetManifestPath(false)
+	if err != nil {
+		return fmt.Errorf("failed to get manifest path for model %s: %w", modelName, err)
+	}
+
+	tarFile, err := os.Create(filePath)
+	if err != nil {
+		return fmt.Errorf("failed to create tar file at %s: %w", filePath, err)
+	}
+	defer tarFile.Close()
+
+	tarWriter := tar.NewWriter(tarFile)
+	defer tarWriter.Close()
+
+	// Add the manifest to the tarball
+	err = handleFile(tarWriter, manifestPath, "manifest.json")
+	if err != nil {
+		return err
+	}
+
+	// Get the home directory
+	home, err := os.UserHomeDir()
+	if err != nil {
+		return fmt.Errorf("failed to get home directory: %w", err)
+	}
+
+	// Create a new spinner
+	s := NewSpinner("Exporting model...")
+	go s.Spin(100 * time.Millisecond)
+
+	// Add each blob to the tarball
+	// Add each blob to the tarball
+	for _, layer := range mf.Layers {
+		blobPath := filepath.Join(home, ".ollama/models/blobs", layer.Digest)
+		err = handleFile(tarWriter, blobPath, "blobs/"+layer.Digest)
+		if err != nil {
+			s.Stop()
+			return err
+		}
+	}
+	s.Stop()
+	fmt.Println("Export completed successfully")
+	return nil
+}
+
+func ImportHandler(cmd *cobra.Command, args []string) error {
+	if len(args) != 2 {
+		return errors.New("incorrect number of arguments. Expected MODELNAME:TAG and FILEPATH")
+	}
+
+	modelTag := strings.Split(args[0], ":")
+	if len(modelTag) != 2 {
+		return errors.New("incorrect format for MODELNAME:TAG. It should be in the format 'model:tag'")
+	}
+
+	newModelName := modelTag[0]
+	newVariant := modelTag[1]
+	filePath := args[1]
+
+	// Open the tar file
+	tarFile, err := os.Open(filePath)
+	if err != nil {
+		return fmt.Errorf("failed to open tar file at %s: %w", filePath, err)
+	}
+	defer tarFile.Close()
+
+	// Create a tar reader
+	tarReader := tar.NewReader(tarFile)
+
+	// Get the home directory
+	home, err := os.UserHomeDir()
+	if err != nil {
+		return fmt.Errorf("failed to get home directory: %w", err)
+	}
+
+	// Create a new spinner
+	s := NewSpinner("Importing model...")
+	go s.Spin(100 * time.Millisecond)
+
+	// Extract each file from the tarball
+	for {
+		header, err := tarReader.Next()
+		if err == io.EOF {
+			break
+		} else if err != nil {
+			s.Stop()
+			return fmt.Errorf("failed to read next file from tarball: %w", err)
+		}
+
+		// Derive the file path based on whether the file is a blob or the manifest
+		var filePath string
+		if strings.HasPrefix(header.Name, "blobs/") {
+			blobName := strings.TrimPrefix(header.Name, "blobs/")
+			// This is a blob, place it in the blobs directory
+			filePath = filepath.Join(home, ".ollama/models/blobs", blobName)
+		} else if header.Name == "manifest.json" {
+			// This is the manifest, place it in the manifests directory and rename it to the variant
+			filePath = filepath.Join(home, ".ollama/models/manifests/registry.ollama.ai/library", newModelName, newVariant)
+		} else {
+			// Unexpected file, skip it
```

Comment:
Not necessary, but I like to log cases like this, just in case it causes some issues in the future it will make things easier to track down.